internal/nodes/avi_model_l4_translator.go

```diff
@@ -149,6 +149,11 @@ func (o *AviObjectGraph) ConstructAviL4PolPoolNodes(svcObj *corev1.Service, vsNo
 			if servers := PopulateServersForNPL(poolNode, svcObj.ObjectMeta.Namespace, svcObj.ObjectMeta.Name, false, key); servers != nil {
 				poolNode.Servers = servers
 			}
+		} else if _, ok := svcObj.GetAnnotations()[lib.SkipNodePortAnnotation]; ok {
```

Comment:
May be we can change this condition to 
ok || serviceType != lib.NodePort  
and remove the last else condition.